ReactNativeClient/lib/services/ExternalEditWatcher.js

```diff
@@ -211,12 +195,52 @@ class ExternalEditWatcher {
 					clearInterval(iid);
 					reject(wrapError(error));
 				});
+				subProcess.on('close', onClose);
 			} catch (error) {
 				throw wrapError(error);
 			}
 		});
 	}
 
+	async spawnDefault(file, onClose) {
+		// windows
+		let path = '';
+
+		if (shim.isWindows()) {
+			path = 'start';
+		} else if (shim.isMac()) {
+			path = 'open';
+		} else if (shim.isLinux() || shim.isFreeBSD()) {
+			path = 'xdg-open';
+		} else {
+			// Fallback on the electron open method
+			bridge().openExternal(`file://${file}`);
+		}
+
+		return this.spawn(path, [file], {}, onClose);
+	}
```

Comment:
Any reason why we need to call `start`, `open`, etc.? Isn't it already what Electron openExternal is doing?

Reply:
I think the logic is not right. If it does the fallback it will also call spawn with an empty path.

Reply:
This code is meant to duplicate the behavior of openExternal with one additional feature, it will actually track close events (that and it get's the benefits of error tracking that spawnCommand already had).

Reply:
> I think the logic is not right. If it does the fallback it will also call spawn with an empty path.

Nice catch! That's my bad, I'll fix that.

Reply:
Actually fixed now!